packages/agent/src/certificate.ts

```diff
@@ -122,7 +122,7 @@ function isBufferEqual(a: ArrayBuffer, b: ArrayBuffer): boolean {
   return true;
 }
 
-type VerifyFunc = (pk: Uint8Array, sig: Uint8Array, msg: Uint8Array) => Promise<boolean>;
+type VerifyFunc = (pk: Uint8Array, sig: Uint8Array, msg: Uint8Array) => Promise<boolean> | boolean;
```

Comment:
Is this  (`=> Promise<boolean> | boolean`) for backwards compatibility?  An approach I've seen elsewhere is to return a completed future in cases like this.

Reply:
It's to allow an interface to have the option to make a call synchronously. The new library is synchronous, and the Promise is the backwards compatibility